kotlin/workflow-runtime/src/main/java/com/squareup/workflow/WorkflowHost.kt

```diff
@@ -149,9 +150,7 @@ internal fun <I : Any, O : Any, R : Any> WorkflowNode<I, *, O, R>.start(
   } catch (e: Throwable) {
     // For some reason the exception gets masked if we don't explicitly pass it to cancel the
     // producer coroutine ourselves here.
-    // TODO https://github.com/square/workflow/issues/188 Stop using parameterized cancel.
-    @Suppress("DEPRECATION")
-    coroutineContext.cancel(e)
+    coroutineContext.cancel(if (e is CancellationException) e else CancellationException(null, e))
```

Comment:
Sure this is still necessary? Wondering if the situation described in your comment was fixed.

Reply:
This `cancel` method only takes `CancellationException`s now, so just trying to not create more exceptions than necessary.